config/config.go

```diff
@@ -329,10 +341,24 @@ func (conf *TerraformConfig) GetAfterHooks() []Hook {
 	return conf.AfterHooks
 }
 
+func (conf *TerraformConfig) GetErrorHooks() []ErrorHook {
+	if conf == nil {
+		return nil
+	}
+
+	return conf.ErrorHooks
+}
+
 func (conf *TerraformConfig) ValidateHooks() error {
-	allHooks := append(conf.GetBeforeHooks(), conf.GetAfterHooks()...)
+	beforeAndAfterHooks := append(conf.GetBeforeHooks(), conf.GetAfterHooks()...)
+
+	for _, curHook := range beforeAndAfterHooks {
+		if len(curHook.Execute) < 1 || curHook.Execute[0] == "" {
+			return InvalidArgError(fmt.Sprintf("Error with hook %s. Need at least one non-empty argument in 'execute'.", curHook.Name))
+		}
+	}
 
-	for _, curHook := range allHooks {
+	for _, curHook := range conf.GetErrorHooks() {
```

Comment:
NIT: reduce this to the following for DRY:

```go
beforeAndAfterHooks := append(conf.GetBeforeHooks(), conf.GetAfterHooks()...)
allHooks := append(beforeAndAfterHooks, conf.GetErrorHooks()...)
for _, curHook := range allHooks {
```

Reply:
won't work, it will probably fail with error, as `ErrorHook` is a different type. The reason for creating different type of hook as `OnErrors` attribute doesn't need to be in before and after hooks. 

```
cannot use conf.GetErrorHooks() (type []ErrorHook) as type []Hook in append
```

I guess the instead of creating new type `ErrorHook` I will have to use the same type, only disadvantage to that will be OnErrors attribute won't be of any use for other two hooks (i.e before and after we can just add validation I guess).

Reply:
Ah that's right makes sense. You can keep it as is for now then!

Reply:
I realised `mergeHooks` and `getIndexOfHookWithName` need to be duplicated too, any suggestions for better design/way to do it. Sorry I my relatively new to Golang

Reply:
It's out of scope of this PR, but the way to handle this is to define a function `validate` on the `Hook` and `ErrorHook` type, and then define an `interface`. Then, you can construct a list of the interface type, and call the `validate` function in the loop.